lib/Doctrine/DBAL/Schema/Table.php

```diff
@@ -616,29 +616,63 @@ public function removeForeignKey($constraintName)
     }
 
     /**
+     * Returns ordered list of columns (primary keys are first, then foreign keys, then the rest)
      * @return Column[]
      */
     public function getColumns()
     {
-        $columns = $this->_columns;
+        return array_merge($this->getPrimaryKeyColumns(), $this->getForeignKeyColumns(), $this->_columns);
```

Comment:
Using `$this->getPrimaryKeyColumns()` in this method will fail if there are no primary keys, and the BC break in `getPrimaryKeyColumns` is reverted (exception throwing on empty PK)